lib/Parse/ParseDecl.cpp

```diff
@@ -3716,6 +3717,18 @@ Parser::parseDecl(ParseDeclOptions Flags,
           Handler(quoteDecl);
         }
       }
+
+      // Set original declaration in `@differentiable` attributes.
```

Comment:
Could you rearrange this to have `for (auto *attr : D->getAttrs().getAttributes<DifferentiableAttr>())` as the outer loop, and assert in the body that we do find an AFD? This will prevent cases with a null original function from slipping through parsing.

Reply:
Thanks for the suggestion! To requestify `@differentiable` attribute parameter indices resolution, I actually found it important/easier to store the original declaration `Decl *` in `DifferentiableAttr`. This helps ensure that the original declaration is set.

Done in https://github.com/apple/swift/pull/27985/commits/0c62032e648d8d49f8110171b171ae19904d21da.